File: lifeagent-frontend/src/types/document.ts
// 与后端 app/schemas/document.py 对齐（Backend API contract is the source of truth）。

export type FileType = 'pdf' | 'txt' | 'markdown'

export type DocumentType =
  | 'contract'
  | 'purchase_record'
  | 'warranty'
  | 'manual'
  | 'note'
  | 'other'

export type DocumentStatus = 'uploaded' | 'processing' | 'completed' | 'failed'

export type ProcessingStage = 'parsing' | 'chunking' | 'embedding' | 'indexing'

export interface DocumentOut {
  document_id: string
  filename: string
  file_type: FileType
  document_type: DocumentType
  file_size: number
  status: DocumentStatus
  processing_stage: ProcessingStage | null
  error_message: string | null
  created_at: string | null
  updated_at: string | null
}

File: app/schemas/document.py
"""Document request and response schemas."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

DocumentType = Literal[
    "contract",
    "purchase_record",
    "warranty",
    "manual",
    "note",
    "other",
]


class DocumentOut(BaseModel):
    document_id: str
    filename: str
    file_type: str
    document_type: str
    file_size: int
    status: str
    processing_stage: str | None = None
    error_message: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
